interfaces/builtin/custom_device.go

```diff
@@ -155,10 +155,21 @@ func (iface *customDeviceInterface) validateUDevTaggingRule(rule map[string]inte
 			err = iface.validateUDevValue(value)
 			if err == nil {
 				deviceName := value.(string)
-				// furthermore, the kernel name must match the name of one of
-				// the given devices
-				if !strutil.ListContains(devices, "/dev/"+deviceName) {
+				// furthermore, the kernel name must match the basename of one
+				// of the given devices
```

Comment:
I think we want deviceName to either match basename of devicePath or or the full device name, so it means /dev/+deviceName ==  devicePath or filepath.Base(devicePath) == deviceName and not other suffixes. We still want to check check that as a suffix deviceName matches only one devicePath possibly?